packages/compass-components/src/components/document-list/element-editors.tsx

```diff
@@ -0,0 +1,329 @@
+import React, { useMemo } from 'react';
+import type { Element as HadronElementType } from 'hadron-document';
+import type { TypeCastMap } from 'hadron-type-checker';
+import TypeChecker from 'hadron-type-checker';
+import { css, cx } from '@leafygreen-ui/emotion';
+import { uiColors } from '@leafygreen-ui/palette';
+import { spacing } from '@leafygreen-ui/tokens';
+import BSONValue, { hasCustomColor, VALUE_COLOR_BY_TYPE } from '../bson-value';
+import { Tooltip } from '../tooltip';
+
+const editorReset = css({
+  padding: 0,
+  margin: 0,
+  border: 'none',
+  boxShadow: 'none',
+  outline: 'none',
+  backgroundColor: 'transparent',
+});
+
+const editorOutline = css({
+  '&:focus, &:active': {
+    borderRadius: `2px`,
+    boxShadow: `0 0 0 2px ${uiColors.focus}`,
+  },
+});
+
+const editorInvalid = css({
+  backgroundColor: uiColors.red.light2,
+  color: uiColors.red.dark2,
+  '&:focus, &:active': {
+    boxShadow: `0 0 0 2px ${uiColors.red.dark2}`,
+  },
+});
+
+export const KeyEditor: React.FunctionComponent<{
+  editing?: boolean;
+  onEditStart(): void;
+  value: string;
+  valid: boolean;
+  validationMessage: string | null;
+  onChange(newVal: string): void;
+  autoFocus?: boolean;
+}> = ({
+  editing,
+  value,
+  valid,
+  validationMessage,
+  onChange,
+  autoFocus,
+  onEditStart,
+}) => {
+  const width = `${Math.max(value.length, 1)}ch`;
```

Comment:
Ah I think we'll want to add some bounds to this rendering:
<img width="967" alt="Screen Shot 2022-03-23 at 2 17 43 PM" src="https://user-images.githubusercontent.com/1791149/159768800-cb1904b2-d367-4511-8639-0c67404a0366.png">

Reply:
Lol, totally, thanks for spotting!

Reply:
Should be fixed now